src/app/mod.rs

```diff
@@ -171,16 +185,44 @@ impl<S: ProvableStore + 'static> Application for BaseCoinApp<S> {
                 &request.data,
                 path.as_ref(),
                 Height::from(request.height as u64),
+                request.prove,
             ) {
                 // success - implies query was handled by this module, so return response
                 Ok(result) => {
+                    let store = self.store.read().unwrap();
+                    let proof_ops = if request.prove {
+                        let proof = store
+                            .get_proof(
+                                Height::from(request.height as u64),
+                                &"ibc".to_owned().try_into().unwrap(),
+                            )
+                            .unwrap();
+                        let mut buffer = Vec::new();
+                        proof.encode(&mut buffer).unwrap(); // safety - cannot fail since buf is a vector
+
+                        let mut ops = vec![];
+                        if let Some(mut proofs) = result.proof {
+                            ops.append(&mut proofs);
+                        }
+                        ops.push(ProofOp {
+                            r#type: "".to_string(),
+                            // FIXME(hu55a1n1)
+                            key: "ibc".to_string().into_bytes(),
```

Comment:
What does this `FIXME` mean? i.e. how should it be fixed?

Reply:
Nice catch! We must return the module's prefix here as the key instead of hard-coding `ibc`. I'll create an issue for this.

Reply:
https://github.com/informalsystems/basecoin-rs/issues/31